blockchain/config.go

```diff
@@ -137,39 +137,32 @@ func (cfg *Config) ProducerPrivateKey() crypto.PrivateKey {
 
 // SetProducerPrivKey set producer privKey by PrivKeyConfigFile info
 func (cfg *Config) SetProducerPrivKey() error {
-	if cfg.PrivKeyConfigFile == "" {
-		return nil
-	}
-
-	yaml, err := config.NewYAML(config.Expand(os.LookupEnv), config.File(cfg.PrivKeyConfigFile))
-	if err != nil {
-		return errors.Wrap(err, "failed to init private key config")
-	}
-	pc := &privKeyConfig{}
-	if err := yaml.Get(config.Root).Populate(pc); err != nil {
-		return errors.Wrap(err, "failed to unmarshal YAML config to privKeyConfig struct")
-	}
-
-	var loader privKeyLoader
-	switch pc.Method {
+	switch cfg.ProducerPrivKeySchema {
+	case "hex", "":
+		// do nothing
 	case "hashiCorpVault":
-		cli, err := newVaultClient(&pc.VaultConfig)
+		yaml, err := config.NewYAML(config.Expand(os.LookupEnv), config.File(cfg.ProducerPrivKey))
```

Comment:
nit: `cfg.ProducerPrivKey` can be a private key text (hex-string) or a filename (like "/var/data/vault.config", which could be a little confusing